### ISSUE TITLE
siteNav: Styles are not being applied for 'layoutName/navigation.md'

### ISSUE THREAD

**Issue by amad-person** [MarkBind/markbind#739]
**Tell us about your environment**

* **MarkBind Version: 1.19.1**

**What did you do? Please include the actual source code causing the issue.**

1. Create a new MarkBind project (run `markbind init`).
2. Add a new page (e.g. `about.md`) with the following **frontmatter**:

```
<frontmatter>
  layout: default
</frontmatter>
```
3. Update `_markbind/layouts/default/navigation.md` with the site navigation menu to include the new page.

      I copied over the contents of `_markbind/navigation/site-nav.md` and added a new link for my page:
```
<navigation>
* [Home :glyphicon-home:]({{baseUrl}}/index.html)
* [About]({{baseUrl}}/about.html)
</navigation>
```
4. Build the site and serve it.

**What did you expect to happen?**

Expected `about.html` to have styles for `#site-nav` and `#page-content`(specified in `site-nav.css`) applied, as it has a site navigation menu specified in its layout.

**What actually happened? Please include the actual, raw output.**

This is how `about.html` looks like:

![screenshot_2019-03-02 about](https://user-images.githubusercontent.com/22164211/53679902-5bc53c80-3d0e-11e9-9faf-1fde28529b63.png)

DevTools showing CSS rules for ~`#site-nav`~ `#page-content` (but site-nav has no styles too) for `about.html`:

<img width="750" alt="devtools showing css for about.html" src="https://user-images.githubusercontent.com/22164211/53679973-587e8080-3d0f-11e9-94fd-9abd214e4018.png">

Meanwhile, `index.html` looks like this:

![screenshot_2019-03-02 hello world](https://user-images.githubusercontent.com/22164211/53679912-7ac3ce80-3d0e-11e9-86b1-25764052eb89.png)

`index.md` has the following **frontmatter**:

```
<frontmatter>
  title: "Hello World"
  footer: footer.md
  siteNav: site-nav.md
</frontmatter>
```

If there is a siteNav attribute present in the **frontmatter**, the styles are being applied. 
DevTools showing CSS rules for `#site-nav` for `index.html`:

<img width="750" alt="devtools showing css for index.html" src="https://user-images.githubusercontent.com/22164211/53679998-ac896500-3d0f-11e9-9dcb-b9cb7484b31e.png">

Is the user supposed to add the IDs manually in the files in for the styles to be applied? I couldn't find any information in the [Page Layouts](https://markbind.org/userGuide/tweakingThePageStructure.html#page-layouts) documentation on this.

**Comment by jamos-tay** [comment]
Ah I think there's a bug with the site template `page.ejs`:

```
    <link rel="stylesheet" href="<%- asset.layoutStyle %>">
    <% if (siteNav) { %><link rel="stylesheet" href="<%- asset.siteNavCss %>"><% } %> <--
    <% if (pageNav) { %><link rel="stylesheet" href="<%- asset.pageNavCss %>"><% } %>
```

The site nav css is only added if a siteNav is provided in the front matter. It should probably be changed to also include it if one is present in the layout.

**Comment by Chng-Zhi-Xuan** [comment]
Duplicate of #528

**Comment by Chng-Zhi-Xuan** [comment]
Issue resolved by #841
